# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.h
# Commit: 5b8986808a06
# Full Hash: 5b8986808a068f773be753351212df596f9ba195
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-12-01 16:18:29
# Description:
#   Bug 1752856 - Null CheckedUnsafePtr<WorkerPrivate> for members of WorkerGlobalScope when releaseing WorkerPrivate. r=dom-worker-reviewers,asuth,smaug
#   
#   In some unusual cases, WorkerGlobalScope could live longer than WorkerPrivate since some other objects hold a reference on WorkerGlobalScope. We try to use CC/GC to release WorkerGlobalScope and force disconnect them from the WorkerPrivate before releasing WorkerPrivate. https://searchfox.org/mozilla-central/rev/83b86005c6913c2062419efb8aabdf2e683aa47f/dom/workers/RuntimeService.cpp#2068-2109
#   
#   The previous implementation focused on nulling the WorkerGlobalScopeBase::mWorkerPrivate. However, it is not enough since WorkerGlobalScope's member could possibly hold a CheckedUnsafePtr<WorkerPrivate>, i.e. WorkerGlobalScope::mPerformance.
# ==============================================================================

diff -r 3c4cb51b8c1d -r 5b8986808a06 dom/workers/WorkerScope.h
--- a/dom/workers/WorkerScope.h	Wed Nov 30 14:49:19 2022 +0000
+++ b/dom/workers/WorkerScope.h	Wed Nov 30 14:27:51 2022 +0000
@@ -19,6 +19,7 @@
 #include "mozilla/dom/AnimationFrameProvider.h"
 #include "mozilla/dom/ImageBitmapBinding.h"
 #include "mozilla/dom/ImageBitmapSource.h"
+#include "mozilla/dom/PerformanceWorker.h"
 #include "mozilla/dom/SafeRefPtr.h"
 #include "mozilla/dom/WorkerPrivate.h"
 #include "nsCOMPtr.h"
